src/App.js

```diff
@@ -1,38 +1,87 @@
 import React, { Component } from 'react';
 import './App.css';
 
+const BlockType = {
+  Heading: 'heading',
+  Text: 'text',
+  Image: 'image',
+  Blockquote: 'blockquote',
+};
+
 class App extends Component {
   state = {
     blocks: [
-      { type: 'heading', text: 'hello world' },
-      { type: 'text', text: 'another string' },
-      { type: 'image', url: 'http://fpoimg.com/300x300', text: 'FPO image' },
-      { type: 'blockQuote', text: 'this is a quote' },
-    ].map(this.withKeys),
+      { type: BlockType.Heading, text: 'hello world' },
+      { type: BlockType.Text, text: 'another string' },
+      {
+        type: BlockType.Image,
+        url: 'http://fpoimg.com/300x300',
+        text: 'FPO image',
+      },
+      { type: BlockType.Blockquote, text: 'this is a quote' },
+    ].map(this.toObjectsWithKeys),
   };
 
-  withKeys(block, index) {
+  toObjectsWithKeys(block, index) {
     const key = btoa(`${index}: ${JSON.stringify(block)}`);
     return { key, ...block };
   }
 
+  // step 2: refactor the addtype functions into generic add block function where you just pass in the block.type as an argument
```

Comment:
Hi there. First of all, everything here works and looks great. This PR doesn't have any errors, and it fixes #6. So good job on all that. The only thing "wrong" is this `addType` function and I'd call it more just like "unfinished". I think you went down a weird side road here and I think it's because the comment we wrote down isn't very clear. We wrote it at the end of a long session, so maybe we just got lazy/tired.

What I meant by `refactor the addtype functions` was `refactor the addHeading, addText, addImage, and addBlockquote functions` (where they all have that `add<Type>` semantic in common) into a _single_ generic `addBlock` function that takes a block type (one of the `BlockType` enum values) as an argument.

The reason we want to refactor here is that you currently have 4 "add" functions that do basically the same thing (pseudo code):

```js
add<BlockType> = () => {
  const new<BlockType> = this.toObjectsWithKeys(
    { type: BlockType.<Type>, text: 'testing' },
    this.state.blocks.length,
  );
  this.setState({ blocks: [...this.state.blocks, new<BlockType> });
};
```

Having this content repeated 4 times violates the DRY principle of software development (Don't Repeat Yourself). It's probably not a super big deal now, but if you have to change the "shape" of a block object later, you'll have to change 4 functions for adding each block type. It'd be great to have one function that could add any type of block. Here's some hints/gotchas to get you started:

```js
addBlock = type => {
  // create your block here and set state like you have before
  // remember you can use { type, /* etc. */ } as a shorthand for { type: type, /* etc. */ }
  // a good way to get unique text for each block is use { text: `${new Date().toLocaleString()}` }
  // think about how you want to handle images, they have that 'url' property which is different than the rest
}
```

To start testing the new `addBlock` in your running app, maybe change the `addHeading` function to look like this:

```js
addHeading = () => this.addBlock(BlockType.Heading);
```